### ISSUE TITLE
Janet API  comments

### ISSUE THREAD

**Issue by bakpakin** [sepisoad/jurl#1]
Browsing through the code,  thought I would answer some questions about the Janet API

* in `easy_setopt`, use -1 in `janet_arity` to ignore min or max arity. (`janet_arity(argc, 2, 1000)` becomes `janet_arity(argc, 2, -1)`)
* In opts.c, the function `opts_set` has some issues when called from `easy_setopt`. The `argv` pointer is only temporary, so leaving pointers to it for callback functions is not going to work often. It seems to work in the test because that part of the stack is never overwritten, but in most programs it won't work.

A really simple fix (that will leak memory, but so be it) is to put arguments in a tuple, reference it so it will not be collected, and then use that instead of argv.

easy.c - old code:
```c
  for (int32_t idx = 1; idx < argc; idx+=2){    
    opts_set(curl, argv + idx, argv + idx + 1);
  }
```

easy.c - patched
```c
  const Janet *argv2 = janet_tuple_n(argv, argc);
  janet_gcroot(janet_wrap_tuple(argv2));
  for (int32_t idx = 1; idx < argc; idx+=2){    
    opts_set(curl, argv2 + idx, argv2 + idx + 1);
  }
```

This won't invoke use after free issues, but the tuple argv2 will never be freed. Using an abstract type would be a good solution that would not leak memory.

**Comment by sepisoad** [comment]
@pepe you can also take care of this issue, I would appreciate

**Comment by pepe** [comment]
I will do it.

**Comment by pepe** [comment]
They were already fixed in #2 and can be closed.

**Comment by sepisoad** [comment]
awesome
